torch/nn/parallel/distributed.py

```diff
@@ -272,6 +273,8 @@ def __init__(self, module, device_ids=None,
         self.module = module
         self.broadcast_buffers = broadcast_buffers
         self.find_unused_parameters = find_unused_parameters
+        self.require_backward_grad_sync = True
+        self.require_forward_param_sync = True
```

Comment:
I'm not sure if `DistributedDataParallel` is picklable, but if it is, then you should add a `__setstate__` that adds those two attributes, because otherwise people who load older checkpoints will get missing attribute errors.